### ISSUE TITLE
Crash during generation of new game map after fresh install

### ISSUE THREAD

**Issue by maxigaz** [jamespetts/simutrans-extended#147]
I've just installed `simutrans-extended` from the AUR on Manjaro Linux for the first time. I've tried it with two different paks; in both cases, the game crashes while the map is being generated.

Here’s the terminal output when using pak256 (also installed from AUR, although I had to edit one line in the PKBBUILD to prevent a “folder not found” error from being shown):

```
Use work dir /home/maxigaz/.local/share/simutrans-extended/
Reading low level config data ...
Parsed simuconf.tab for directory layout; multiuser = 1
parse_simuconf() in program dir (config/simuconf.tab): Reading simuconf.tab successful!
SDL Driver: x11
Preparing display ...
Loading font 'font/prop.fnt'
font/prop.fnt successfully loaded as old format prop font!
Init done.
modal_dialogue( sel, magic_none, NULL, empty_objfilename );
Pak found: pak256-ex/
parse_simuconf() at pak256-ex/config/simuconf.tab: 
Reading simuconf.tab successful!
Reading compatibility sound data ...
Loading BDF font 'cyr.bdf'
Reading city configuration ...
Reading electricity consumption configuration ...
Reading menu configuration ...
Reading object data from pak256-ex/...
Reading menu configuration ...
Reading private car ownership configuration ...
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Error: Cannot open 'wenquanyi_9pt.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'm+10r.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Error: Cannot open 'wenquanyi_9pt.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
modal_dialogue( sel, magic_none, NULL, no_language );
Loading BDF font 'cyr.bdf'
Loading BDF font 'cyr.bdf'
Midi disabled ...
Calculating textures ...done
World destroyed.
parse_simuconf() in pak dir (pak256-ex/config/simuconf.tab) for override of save file: Reading simuconf.t
ab successful!
[1]    53028 floating point exception (core dumped)  simutrans-extended
```

And here’s the output using pak128.german (downloaded from [the website of Simutrans](https://www.simutrans.com/en/paksets/)):

```
Use work dir /home/maxigaz/.local/share/simutrans-extended/
Reading low level config data ...
Parsed simuconf.tab for directory layout; multiuser = 1
parse_simuconf() in program dir (config/simuconf.tab): Reading simuconf.tab successful!
SDL Driver: x11
Preparing display ...
Loading font 'font/prop.fnt'
font/prop.fnt successfully loaded as old format prop font!
Init done.
modal_dialogue( sel, magic_none, NULL, empty_objfilename );
Pak found: pak128.german/
parse_simuconf() at pak128.german/config/simuconf.tab: 
Reading simuconf.tab successful!
Reading compatibility sound data ...
Loading BDF font 'cyr.bdf'
Reading city configuration ...
Reading electricity consumption configuration ...
Reading menu configuration ...
Reading object data from pak128.german/...
[1]    58044 segmentation fault (core dumped)  simutrans-extended
```

P.S.: Here are the last few lines when `-debug 5` was used with pak128.german (if it helps):

```
Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (5,1) Flag 83

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (2,2) Flag 83

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (3,2) Flag 72

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (4,2) Flag 72

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (5,2) Flag 83

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (4,3) Flag 72

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (3,4) Flag 115

Message: stadt_t::cityrules_init():	Road-Rule 36: Pos (4,4) Flag 115

Reading electricity consumption configuration ...
Message: stadt_t::electricity_consumption_init():	Error opening config/electricity.tab.
Will use default value.
Reading menu configuration ...
ERROR: create_general_tool():	cannot satisfy request for general_tool[40]!
For help with this error or to file a bug report please see the Simutrans forum:
http://forum.simutrans.com
Warning: create_simple_tool():	deprecated tool [27] requested
Reading object data from pak128.german/...
Message: ground_desc_t::register_desc():	setting raster width to 128
[1]    56960 segmentation fault (core dumped)  simutrans-extended -log simu-pak256.log -debug 5
```

(I also noticed the `-log` switch can be used for saving the log in a file, but I couldn’t find `simu.log` anywhere, that’s why I pasted the terminal output above instead.)

**Comment by Roboron3042** [comment]
Sorry, but Simutrans-Extended updates the code everyday, and there's not a stable release, so it may be that you accidentally installed a borked version.

If you update the game and try again, you probably won't be able to reproduce this crash :-)

**Comment by maxigaz** [comment]
Thank you for the reply!

Actually, around the time I wrote the report, I tried several different updates, and the results were the same. Also, in a forum thread that seemed to be about a similar issue, I found a reply from the developer asking for running it through a development tool. (Probably for backtrace?) That appeared too complicated for me and made me realise that the log I posted here was probably useless, so I gave up.

Anyway, all that was back in March, so the issue may have been fixed since then. I might try again some day. For now, I’m closing the issue.
